src/build_helper/src/fs/mod.rs

```diff
@@ -31,25 +31,47 @@ where
 ///
 /// FIXME: this implementation is insufficiently robust to replace bootstrap's clean `rm_rf`
 /// implementation:
-///
-/// - This implementation currently does not perform retries.
 #[track_caller]
 pub fn recursive_remove<P: AsRef<Path>>(path: P) -> io::Result<()> {
     let path = path.as_ref();
-    let metadata = fs::symlink_metadata(path)?;
-    #[cfg(windows)]
-    let is_dir_like = |meta: &fs::Metadata| {
-        use std::os::windows::fs::FileTypeExt;
-        meta.is_dir() || meta.file_type().is_symlink_dir()
-    };
-    #[cfg(not(windows))]
-    let is_dir_like = fs::Metadata::is_dir;
 
-    if is_dir_like(&metadata) {
-        fs::remove_dir_all(path)
-    } else {
-        try_remove_op_set_perms(fs::remove_file, path, metadata)
+    for retry in 0..5 {
+        match fs::symlink_metadata(path) {
```

Comment:
Problem: this retry logic is wrong:

- We should not be retrying `symlink_metadata`, it's the *removal* that I'd like to retry. Otherwise, on Windows, removal operations AFAIK can *mark* a file for removal but not necessarily remove it immediately.
- In the actual removal result inspection, the retry condition should inspect the error kind, and *only* retry removal if the error condition is *not* `io::ErrorKind::NotFound` -- if error kind is `io::ErrorKind::NotFound`, return success.
- There is no delay window between consecutive retries. I'd add a very short delay between each retry to give e.g. malware scanners an opportunity to release its handle on the given files in that case might help.